test/integration/conftest.py

```diff
@@ -129,10 +129,9 @@ def is_feature_store_available(region) -> bool:
         try:
             sagemaker_client.list_feature_groups()
             return True
-        except botocore.exceptions.EndpointConnectionError:
-            logging.info(
-                "Caught EndpointConnectionError when checking the feature store availability,"
-                " retried %d times already",
+        except Exception:
```

Comment:
Looks like EndpointConnectionError is not a concrete indicator for whether a service is available or not in a region. Client error can be thrown as well.

--> Can we catch both the specific exception rather than give it very generic exception to narrow down the issue. retry will apply to all exception.

Reply:
I need to get some context. What is the intention of this test? Should we not have a config with regions where FS is not available and then skip the tests in those regions?

Reply:
After synced with Ajay, I turned to use static config for validating if FS is available or not.